pkg/apis/core/v1/defaults_test.go

```diff
@@ -3182,15 +3183,26 @@ func TestSetDefaultResizePolicy(t *testing.T) {
 			},
 		},
 	} {
-		t.Run(desc, func(t *testing.T) {
-			testPod := v1.Pod{}
-			testPod.Spec.Containers = append(testPod.Spec.Containers, tc.testContainer)
-			output := roundTrip(t, runtime.Object(&testPod))
-			pod2 := output.(*v1.Pod)
-			if !cmp.Equal(pod2.Spec.Containers[0].ResizePolicy, tc.expectedResizePolicy) {
-				t.Errorf("expected resize policy %+v, but got %+v", tc.expectedResizePolicy, pod2.Spec.Containers[0].ResizePolicy)
-			}
-		})
+		for _, isSidecarContainer := range []bool{true, false} {
+			t.Run(desc, func(t *testing.T) {
+				testPod := v1.Pod{}
+				if isSidecarContainer {
+					tc.testContainer.RestartPolicy = &restartAlways
+					testPod.Spec.InitContainers = append(testPod.Spec.InitContainers, tc.testContainer)
+				} else {
+					testPod.Spec.Containers = append(testPod.Spec.Containers, tc.testContainer)
+				}
```

Comment:
I susggest always adding the container to init containers, but only setting the RestartPolicy if `isSidecarContainer`
```suggestion
				testPod.Spec.Containers = append(testPod.Spec.Containers, tc.testContainer)
				initContainer := *tc.testContainer.DeepCopy()
				if isSidecarContainer {
					initContainer.RestartPolicy = &restartAlways
				}
				testPod.Spec.InitContainers = append(testPod.Spec.InitContainers, initContainer)
```

then below, verify that no defaults are set on the init container if it's not a sidecar.